designsafe/apps/api/agave/filemanager/publications.py

```diff
@@ -0,0 +1,107 @@
+"""
+.. module: portal.apps.search.api.managers.private_data_search
+   :synopsis: Manager handling My Data searches.
+"""
+
+from __future__ import unicode_literals, absolute_import
+import logging
+import datetime
+from designsafe.apps.api.search.searchmanager.base import BaseSearchManager
+from designsafe.apps.data.models.elasticsearch import IndexedFile
+from elasticsearch_dsl import Q, Search, Index
+from django.conf import settings
+from designsafe.libs.elasticsearch.docs.publications import BaseESPublication
+from designsafe.libs.elasticsearch.docs.publication_legacy import BaseESPublicationLegacy
+from designsafe.apps.api.agave.filemanager.agave import  AgaveFileManager
+logger = logging.getLogger(__name__)
+
+
+class PublicationsManager(AgaveFileManager):
+    """ File manager for listing publications.
+    """
+
+    @property
+    def requires_auth(self):
+        """Whether it should check for an authenticated user.
+
+        If this is a public data file manager, it should return False.
+        """
+        return False
+
+    def construct_query(self, system=None, file_path=None):
+
+        published_index_name = Index('des-publications').get_alias().keys()[0]
+        legacy_index_name = Index('des-publications_legacy').get_alias().keys()[0]
+
+        published_query = Q(
+            'bool',
+            must=[
+                Q('bool', should=[
+                    Q({'term': {'_index': published_index_name}}),
+                    Q({'term': {'_index': legacy_index_name}})
+                ])
+            ],
+            must_not=[
+                Q('term', status='unpublished'),
+                Q('term', status='saved')
+            ]
+        )
+
+        return published_query
+
+    def listing(self, system=None, file_path=None, offset=0, limit=100, **kwargs):
+        """Wraps the search result in a BaseFile object for serializtion."""
+        query = self.construct_query(system, file_path)
+        listing_search = Search()
+        listing_search = listing_search.query(query).sort(
+            '_index',
+            {'project._exact': {'order':'asc', 'unmapped_type': 'keyword'}},
+            {'created': {'order': 'desc', 'unmapped_type': 'long'}}
+        )
+        listing_search = listing_search.extra(from_=offset, size=limit)
+
+        res = listing_search.execute()
+        children = []
+        for hit in res:
+            try:
+                pid = hit.projectId
```

Comment:
I don't think we are using `pid` in here.

Reply:
I'm trying to distinguish between DES and NEES publications by checking for a `projectId` attribute- I'll refactor to make that more readable.